test/k8sT/Policies.go

```diff
@@ -281,5 +281,240 @@ var _ = Describe("K8sPolicyTest", func() {
 			helpers.DefaultNamespace, appPods[helpers.App3], fmt.Sprintf("curl --fail -s http://%s/public", clusterIP))
 		Expect(err).Should(BeNil())
 	}, 500)
+})
+
+var _ = Describe("K8sPolicyTestAcrossNamespaces", func() {
+
+	var initialized bool
+	var kubectl *helpers.Kubectl
+	var logger *logrus.Entry
+	var path string
+
+	initialize := func() {
+		if initialized == true {
+			return
+		}
+
+		logger = log.WithFields(logrus.Fields{"testName": "K8sPolicyTestAcrossNamespaces"})
+		logger.Info("Starting")
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+		path = kubectl.ManifestGet("cilium_ds.yaml")
+		kubectl.Apply(path)
+		status, err := kubectl.WaitforPods(helpers.KubeSystemNamespace, "-l k8s-app=cilium", 300)
+		Expect(status).Should(BeTrue())
+		Expect(err).Should(BeNil())
+		initialized = true
+	}
+
+	BeforeEach(func() {
+		initialize()
+	})
+
+	AfterEach(func() {
+		if CurrentGinkgoTestDescription().Failed {
+			ciliumPod, _ := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+			kubectl.CiliumReport(helpers.KubeSystemNamespace, ciliumPod, []string{
+				"cilium bpf tunnel list",
+				"cilium endpoint list"})
+		}
+	})
+	It("Policies Across Namespaces", func() {
+
+		namespace := "namespace"
+		qaNs := "qa"
+		developmentNs := "development"
+		podNameFilter := "{.items[*].metadata.name}"
+
+		// formatLabelArgument formats the provided key-value pairs as labels for use in
+		// querying Kubernetes.
+		formatLabelArgument := func(firstKey, firstValue string, nextLabels ...string) string {
+			baseString := fmt.Sprintf("-l %s=%s", firstKey, firstValue)
+			if nextLabels == nil {
+				return baseString
+			} else if len(nextLabels)%2 != 0 {
+				panic("must provide even number of arguments for label key-value pairings")
+			} else {
+				for i := 0; i < len(nextLabels); i += 2 {
+					baseString = fmt.Sprintf("%s,%s=%s", baseString, nextLabels[i], nextLabels[i+1])
+				}
+			}
+			return baseString
+		}
+
+		testConnectivity := func(backendIP, frontendPod string) {
```

Comment:
Can you swap the `backend` with `frontend`? We usually test `from -> to`

Reply:
Fixed.